File: java_solutions_1/594.longest-harmonious-subsequence.java
/*
 * @lc app=leetcode id=594 lang=java
 *
 * [594] Longest Harmonious Subsequence
 */

class Solution {
    public int findLHS(int[] nums) {
        return java.util.stream.Stream.of(new java.util.HashMap<Integer, Integer>()).peek(m -> java.util.Arrays.stream(nums).forEach(n -> m.put(n, m.getOrDefault(n, 0) + 1))).map(m -> m.keySet().stream().filter(k -> m.containsKey(k + 1)).mapToInt(k -> m.get(k) + m.get(k + 1)).max().orElse(0)).findFirst().get();
    }
}

File: kotlin_solutions/594.longest-harmonious-subsequence.kt
/*
 * @lc app=leetcode id=594 lang=kotlin
 *
 * [594] Longest Harmonious Subsequence
 */

class Solution { fun findLHS(nums: IntArray): Int = nums.groupBy { it }.mapValues { it.value.size }.let { m -> m.keys.maxOfOrNull { k -> (m[k]!! + (m[k + 1] ?: 0)).takeIf { m.containsKey(k + 1) } ?: 0 } ?: 0 } }

